daml/Marketplace/Issuance/AssetDescription.daml

```diff
@@ -47,19 +45,10 @@ template AssetDescription
 template Index
   with
     assetLabel: Text
+    claims: Claims
```

Comment:
Yes, this is what I had in mind.

Looking at it now, I'm wondering whether we should have an `Asset` instead of the separate label/version. Not sure. What do you think?

Reply:
Is it not enough to have an Id (part of the Asset)?

Reply:
See also comments above; we could consider `ContractId AssetDescription` to replace `assetLabel, version, descriptionSingatories`. This way we can only create an `Index` if the underlying `AssetDescription` exists first.

Reply:
I understand that there should be an asset description contract Id. But with that, we still cannot eliminate assetLabel, descriptionSingatories - those are parts of the key (the latter is also the value used for signatory).
Version needs to be here as well, I guess (key).

So as far as I can understand signatories + version + label = Id, or not @lucianojoublanc-da ? What am I missing?

Reply:
> those are parts of the key

Oh you're right. I don't think it's possible to get around that.

So perhaps just have an `Id` embedded there instead of individual

> signatories + version + label = Id

fields?

Reply:
Yes exactly

Reply:
It looks good!